probes/pinsDependencies/impl_test.go

```diff
@@ -0,0 +1,669 @@
+						{
+							Location: &checker.File{
+								Snippet: "actions/checkout@v2",
+							},
+							Type:   checker.DependencyUseTypeGHAction,
+							Pinned: asBoolPointer(false),
+						},
+						{
+							Location: &checker.File{
+								Snippet: "other/checkout@a81bbbf8298c0fa03ea29cdc473d45769f953675",
+							},
+							Type:   checker.DependencyUseTypeGHAction,
+							Pinned: asBoolPointer(true),
+						},
+					},
+				},
+			},
+			outcomes: []finding.Outcome{
+				finding.OutcomeNegative,
+				finding.OutcomePositive,
+			},
+		},
+		{
+			name: "Skipped objects and dependencies",
+			raw: &checker.RawResults{
+				PinningDependenciesResults: checker.PinningDependenciesData{
+					Dependencies: []checker.Dependency{
+						{
+							Location: &checker.File{},
+							Type:     checker.DependencyUseTypeNpmCommand,
+							Pinned:   asBoolPointer(false),
+						},
+						{
+							Location: &checker.File{},
+							Type:     checker.DependencyUseTypeNpmCommand,
+							Pinned:   asBoolPointer(false),
+						},
+					},
+				},
+			},
+			outcomes: []finding.Outcome{
+				finding.OutcomeNegative,
+				finding.OutcomeNegative,
+			},
+		},
+		{
+			name: "dependency missing Location info and no error message throws error",
+			raw: &checker.RawResults{
+				PinningDependenciesResults: checker.PinningDependenciesData{
+					Dependencies: []checker.Dependency{
+						{
+							Location: nil,
+							Msg:      nil,
+							Type:     checker.DependencyUseTypeNpmCommand,
+							Pinned:   asBoolPointer(true),
+						},
+					},
+				},
+			},
+			err: sce.ErrScorecardInternal,
+		},
+		{
+			name: "dependency missing Location info",
+			raw: &checker.RawResults{
+				PinningDependenciesResults: checker.PinningDependenciesData{
+					Dependencies: []checker.Dependency{
+						{
+							Location: nil,
+							Msg:      &msg,
+							Type:     checker.DependencyUseTypeNpmCommand,
+							Pinned:   asBoolPointer(true),
+						},
+					},
+				},
+			},
+			outcomes: []finding.Outcome{
+				finding.OutcomeNotApplicable,
+			},
+		},
+		{
+			name: "neither location nor msg is nil",
+			raw: &checker.RawResults{
+				PinningDependenciesResults: checker.PinningDependenciesData{
+					Dependencies: []checker.Dependency{
+						{
+							Location: &checker.File{},
+							Msg:      &msg,
+							Type:     checker.DependencyUseTypeNpmCommand,
+							Pinned:   asBoolPointer(true),
+						},
+					},
+				},
+			},
+			outcomes: []finding.Outcome{
+				finding.OutcomeNotApplicable,
+			},
+		},
+		{
+			name: "pinned = nil",
+			raw: &checker.RawResults{
+				PinningDependenciesResults: checker.PinningDependenciesData{
+					Dependencies: []checker.Dependency{
+						{
+							Location: &checker.File{},
+							Msg:      nil,
+							Type:     checker.DependencyUseTypeNpmCommand,
+							Pinned:   nil,
+						},
+					},
+				},
+			},
+			outcomes: []finding.Outcome{
+				finding.OutcomeNotApplicable,
+			},
+		},
+		{
+			name: "2 processing errors",
```

Comment:
nit: describe the purpose, not describe the test.

```
name: "processing errors result in OutcomeError"
```

Reply:
https://github.com/ossf/scorecard/pull/3829/commits/a6947e6ca7b69a7153ab857cd34a3f1f2feb790f